integrationtests/agent/adoption_test.go

```diff
@@ -0,0 +1,387 @@
+				condition.Reason == "Error" &&
+				strings.Contains(condition.Message, "not owned by us") {
+				return nil
+			}
+		}
+		return fmt.Errorf("does not match expected condition")
+	}
+
+	bundleDeploymentReady := func(bd *v1alpha1.BundleDeployment) error {
+		for _, condition := range bd.Status.Conditions {
+			if condition.Type == v1alpha1.BundleDeploymentConditionReady &&
+				condition.Status == corev1.ConditionTrue {
+				return nil
+			}
+		}
+		return fmt.Errorf("does not match expected condition")
+	}
+
+	BeforeEach(func() {
+		namespace = createNamespace()
+		DeferCleanup(func() {
+			Expect(k8sClient.Delete(ctx, &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{Name: namespace}})).ToNot(HaveOccurred())
+		})
+		env = &specEnv{namespace: namespace}
+	})
+
+	When("a resource of a bundle deployment is removed", Label("remove-resource"), func() {
+		It("should report the deleted resource as missing", func() {
+			createBundleDeployment("remove-resource", false)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				var err error
+				err = mapPartialMatch(cm.Data, map[string]string{"key": "value"})
+				if err != nil {
+					return err
+				}
+				err = mapPartialMatch(cm.Annotations, map[string]string{
+					"meta.helm.sh/release-name":      "remove-resource",
+					"meta.helm.sh/release-namespace": "",
+					"objectset.rio.cattle.io/id":     "default-remove-resource",
+				})
+				if err != nil {
+					return err
+				}
+				err = mapPartialMatch(cm.Labels, map[string]string{
+					"app.kubernetes.io/managed-by": "Helm",
+					"objectset.rio.cattle.io/hash": "ca7682543199bb801d0c14587a1158d936508160",
+				})
+				if err != nil {
+					return err
+				}
+				return nil
+			})
+			// This is required for the resource to be watched.
+			time.Sleep(5 * time.Second)
+			deleteConfigMap("cm1")
+			assertBundleDeployment("remove-resource", bundleDeploymentResourceMissing)
+		})
+	})
+
+	When("all labels of a resource of a bundle deployment are removed", Label("remove-labels"), func() {
+		It("should report that resource as \"not owned by us\"", func() {
+			createBundleDeployment("remove-metadata", false)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := mapPartialMatch(cm.Data, map[string]string{"key": "value"}); err != nil {
+					return err
+				}
+				if err := mapPartialMatch(cm.Annotations, map[string]string{
+					"meta.helm.sh/release-name":      "remove-metadata",
+					"meta.helm.sh/release-namespace": "",
+					"objectset.rio.cattle.io/id":     "default-remove-metadata",
+				}); err != nil {
+					return err
+				}
+
+				return mapPartialMatch(cm.Labels, map[string]string{
+					"app.kubernetes.io/managed-by": "Helm",
+					"objectset.rio.cattle.io/hash": "0f3e1d9d146fa8b290c0de403881184751430e59",
+				})
+			})
+			changeConfigMap("cm1", func(cm *corev1.ConfigMap) {
+				cm.Labels = map[string]string{}
+			})
+			assertBundleDeployment("remove-metadata", bundleDeploymentNotOwnedByUs)
```

Comment:
Just to be sure I understand this properly: in this case, does it still make sense to expect a bundle deployment not to be ready because the config map is still specified as part of its resources?

In other words, do we expect removing the config map from the bundle deployment (apart from restoring its labels) to be the way to go to enable the bundle deployment to reach ready state?

Reply:
Yes. Do you think it would be better if that was different?